core/modules/replica/tools/qserv-replica-master.cc

```diff
@@ -35,715 +40,322 @@
+     */
+    MasterControllerApp(int argc,
+                        const char* const argv[],
+                        std::string const& description,
+                        std::string const& logger)
+        :   Application(
+                argc,
+                argv,
+                description,
+                true /* injectDatabaseOptions */,
+                true /* boostProtobufVersionCheck */,
+                true /* enableServiceProvider */
+            ),
+            _healthProbeIntervalSec  (60),
+            _replicationIntervalSec  (60),
+            _workerResponseTimeoutSec(60),
+            _workerEvictTimeoutSec   (3600),
+            _qservSyncTimeoutSec     (60),
+            _numReplicas             (0),
+            _numIter                 (0),
+            _purge          (false),
+            _forceQservSync (false),
+            _permanentDelete(false),
+            _failed         (false),
+            _log(LOG_GET(logger)) {
+
+        // Configure the command line parser
+
+        parser().option(
+            "health-probe-interval",
+            "interval (seconds) between iterations of the health monitoring probes",
+            _healthProbeIntervalSec
+        ).option(
+            "replication-interval",
+            "interval (seconds) between running the linear sequence of"
+            " actions: check - fix-up - replicate - re-balance",
+            _replicationIntervalSec
+        ).option(
+            "worker-response-timeout",
+            "maximum number of seconds to wait before giving up"
+            " on worker probes when checking for workers",
+            _workerResponseTimeoutSec
+        ).option(
+            "worker-evict-timeout",
+            "the maximum number of seconds to allow troubles workers to recover"
+            " from the last catastrophic event before evicting them from a cluster",
+            _workerEvictTimeoutSec
+        ).option(
+            "qserv-sync-timeout",
+            "the maximum number of seconds to wait before Qserv workers respond"
+            " to the synchronization requests before bailing out and proceeding"
+            " to the next step in the normal replication sequence. A value which"
+            " differs from 0 would override the corresponding parameter specified"
+            " in the Configuration.",
+            _qservSyncTimeoutSec
+        ).flag(
+            "qserv-sync-force",
+            "the flag which would force Qserv workers to update their list of replicas"
+            " even if some of the chunk replicas were still in use by on-going queries."
+            " This affect replicas to be deleted from the workers during the synchronization"
+            " stages",
+            _forceQservSync
+        ).option(
+            "replicas",
+            "the minimal number of replicas when running the replication phase"
+            " This number if provided and if it's not 0 will override the corresponding value found"
```

Comment:
there could be trouble with the default value changing and this not getting updated. Maybe a `const int kNumDefaultWorkers` _is_ a good idea...

Reply:
The value of this parameter has nothing to do with the number of workers (as per your suggestion for `kNumDefaultWorkers`). Though, I can see your concern - putting a fixed number 0 may not be a good idea here. I will think about how to rephrase the parameter description to make it less confusing.

Reply:
Fixed